lib/Sema/CSSimplify.cpp

```diff
@@ -3802,6 +3802,24 @@ bool ConstraintSystem::repairFailures(
       }
     }
 
+    if (isForCodeCompletion()) {
+      // If the argument contains the code completion location, the user has not
+      // finished typing out this argument yet. Treat the mismatch as valid so
+      // we don't penalize this solution.
+      if (auto *arg = getAsExpr(simplifyLocatorToAnchor(loc))) {
+        // Ignore synthesized args like $match in implicit pattern match
+        // operator calls. Their source location is usually the same as the
+        // other (explicit) argument's so source range containment alone isn't
+        // sufficient.
+        bool isSynthesizedArg = arg->isImplicit() && isa<DeclRefExpr>(arg);
+        SourceRange range = arg->getSourceRange();
+        if (!isSynthesizedArg && range.isValid() &&
+            Context.SourceMgr.rangeContainsCodeCompletionLoc(range) &&
```

Comment:
I think it would be good to get `range.isValid() &&
            Context.SourceMgr.rangeContainsCodeCompletionLoc(range)` condition extracted into a separate method on constraint system, something like `ConstraintSystem::containsCodeCompletionToken(Expr *expr)` since that's a pattern we here and in CSGen.